oak_crypto/src/tests.rs

```diff
@@ -265,3 +265,29 @@ async fn test_async_encryptor() {
     assert_eq!(TEST_RESPONSE_MESSAGE, decrypted_response);
     assert_eq!(TEST_RESPONSE_ASSOCIATED_DATA, response_associated_data);
 }
+
+const _TEST_SIGNATURE_MESSAGE_ONE_: &[u8] = b"Dogs are the best";
+const _TEST_SIGNATURE_MESSAGE_TWO_: &[u8] = b"Cats are even better";
+
+use crate::{signer::Signer, verifier::Verifier};
+
+#[test]
+fn test_signer_and_verifier() {
+    let (signing_key_one, verifying_key_one) = oak_dice::cert::generate_ecdsa_key_pair();
```

Comment:
that's a good point, I agree oak_crypto should not depend on oak_dice at all, but it looks like it does already:

https://github.com/project-oak/oak/blob/ebb0aab17c23b3327d5d97e2ff38639d48f1f7ed/oak_crypto/src/encryptor.rs#L48-L80

Maybe that impl should be moved to oak_dice?

Reply:
> Ideally we need to create a struct called a SigningKey (and also a trait SigningKeyHandle, which will be implemented by our SDKs)

I just recalled that we don't have the SigningKeyHandle abstraction. So maybe just `SigningKey` could be defined in the `oak_crypto`

Reply:
+1 I suspect we just want to create the actual signing key in oak_crypto, and just expose an opaque wrapper around it (not the specific `p256::ecdsa::SigningKey` type) with a sign method that returns the proto that then the attestation verification library will be able to use. WDYT?

ref: https://github.com/project-oak/oak/blob/2a9a18a016c5cd5c4ed3915fc53cf1831818d907/oak_restricted_kernel_sdk/src/dice.rs#L25

Reply:
> +1 I suspect we just want to create the actual signing key in oak_crypto, and just expose an opaque wrapper around it (not the specific p256::ecdsa::SigningKey type) with a sign method that returns the proto that then the attestation verification library will be able to use. WDYT?

I don't think we need to do this, it feels like we may be applying applying C++ patterns to rust to me. 

Let's have that discussion in this thread: https://github.com/project-oak/oak/pull/4683#discussion_r1458842095

Reply:
> that's a good point, I agree oak_crypto should not depend on oak_dice at all, but it looks like it does already

was surprised it depended on it also. If we want to change that, it should be a separate refactoring PR.

Reply:
> I don't think `oak_crypto` should depend on the `oak_dice` create. I.e. `oak_crypto` should have it's own mechanisms for generating keys, similar to
> 
> https://github.com/project-oak/oak/blob/ebb0aab17c23b3327d5d97e2ff38639d48f1f7ed/oak_crypto/src/hpke/mod.rs#L50-L57
> 
> And it's `oak_dice` who should reuse it.

about your last point, oak_dice is used in stage0. If it were to depend on `oak_crypto`, it would bloat stage0 a lot, since `oak_crypto` includes deps for protos among others.

Reply:
Maybe we need a oak_crypto_core or similar. (not sure if it's a good idea though)

Reply:
I think at this point we can just duplicate key generation instead of creating 2 crypto crates. We just need to make sure we are generating keys in the same format

Reply:
That solves the usecase in this PR, but still leaves us with the bit that @tiziano88  mentioned

> that's a good point, I agree oak_crypto should not depend on oak_dice at all, but it looks like it does already:
> 
> https://github.com/project-oak/oak/blob/ebb0aab17c23b3327d5d97e2ff38639d48f1f7ed/oak_crypto/src/encryptor.rs#L48-L80
> 
> Maybe that impl should be moved to oak_dice?